crates/runc-shim/src/cgroup_memory.rs

```diff
@@ -0,0 +1,184 @@
+use containerd_shim::{
+    error::{Error, Result},
+    io_error, other_error,
+};
+use std::path::Path;
+use tokio::{
+    fs::{self, read_to_string, File},
+    io::AsyncReadExt,
+    sync::mpsc::{self, Receiver},
+};
+
+use nix::sys::eventfd;
+use std::os::unix::io::{AsRawFd, FromRawFd};
+
+pub async fn get_path_from_cgorup(pid: u32) -> Result<String> {
+    let proc_path = format!("/proc/{}/cgroup", pid);
+    let path_string = read_to_string(proc_path.clone()).await.map_err(io_error!(
+        e,
+        "open {}.",
+        proc_path.clone()
```

Comment:
you can borrow `&proc_path` in both places instead of cloning.

Reply:
Ok.